File: Android/Lima/app/src/main/java/start/up/tracker/analytics/Principle.java
package start.up.tracker.analytics;

public interface Principle {
    void setStatus(boolean isEnabled);

    boolean getStatus();

    int getTimeToRead();

    String getName();

    void logic();
}

/*
public abstract class Principle {
    private String name;
    //private String description;
    private boolean isEnabled;
    private int timeToRead;

    public void setStatus(boolean isEnabled) {
        this.isEnabled = isEnabled;
    }

    public boolean getStatus() {
        return isEnabled;
    }

    public int getTimeToRead() {
        return timeToRead;
    }

    public String getName() {
        return name;
    }

    abstract public void logic();
}

 */

File: Android/Lima/app/src/main/java/start/up/tracker/analytics/Principle.kt
package start.up.tracker.analytics

import start.up.tracker.entities.Task
import java.util.ArrayList

interface Principle {
    fun setStatus(status: Boolean)
    fun canBeEnabled(activePrinciples: List<Principle>): Boolean
    fun getStatus(): Boolean
    fun setNotifications(notifications: Boolean)
    fun getNotifications(): Boolean
    fun getTimeToRead(): Int?
    fun getReference(): String?
    fun getName(): String?
    fun getIncompatiblePrinciples(): ArrayList<Principle>?
    fun logic()
    suspend fun logicAddTask(task: Task)
    suspend fun logicEditTask(task: Task)
}